### PR TITLE
CAT-2274 GPII: Simple Bricks

### PR DESCRIPTION
**Pull request by Manu91**

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ProjectManager.java [hunk 1]
```diff
@@ -793,6 +798,31 @@ private boolean checkReferencesOfCurrentBrick(Brick currentBrick) {
...
+			if (endBrick == null || endBrick.getIfBeginSimpleBrick() == null
+					|| !endBrick.getIfBeginSimpleBrick().equals(currentBrick)) {
+				Log.d(TAG, "Brick has wrong reference:" + currentSprite + " "
+						+ currentBrick);
+				return false;
+			}
+		} else if (currentBrick instanceof IfLogicBeginSimpleBrick) {
+			IfLogicElseSimpleBrick elseBrick = ((IfLogicBeginSimpleBrick) currentBrick).getIfElseSimpleBrick();
+			IfLogicEndSimpleBrick endBrick = ((IfLogicBeginSimpleBrick) currentBrick).getIfEndSimpleBrick();
+			if (elseBrick == null || endBrick == null || elseBrick.getIfEndSimpleBrick() == null
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/ProjectManager.java]: I know that this is the way it was done before, but this is really hard to read.
At least put the whole simple bricks block and the "normal" if then else block in separate methods

File: catroid/src/main/java/org/catrobat/catroid/ProjectManager.java [hunk 2]
```diff
@@ -866,6 +896,57 @@ private void correctAllNestedReferences(Script script) {
 			bricksWithInvalidReferences.add(brick);
 		}
 
+		ArrayList<IfLogicBeginSimpleBrick> ifBeginSimpleList = new ArrayList<>();
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/ProjectManager.java]: same here.. separate methods would be nice

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicElseSimpleBrick.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	@Override
+	public void initialize() {
+		Log.w(TAG, "Cannot create the IfLogic Bricks from here!");
+	}
+
+	@Override
+	public List<NestingBrick> getAllNestingBrickParts(boolean sorted) {
+		//TODO: handle sorting
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicElseSimpleBrick.java]: is this important?

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicElseSimpleBrick.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+		List<NestingBrick> nestingBrickList = new ArrayList<>();
+		if (sorted) {
+			nestingBrickList.add(ifBeginBrick);
+			nestingBrickList.add(this);
+			nestingBrickList.add(ifEndBrick);
+		} else {
+			//nestingBrickList.add(this);
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicElseSimpleBrick.java]: please remove unused code

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicEndSimpleBrick.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+	@Override
+	public void initialize() {
+		Log.w(TAG, "Cannot create the IfLogic Bricks from here!");
+	}
+
+	@Override
+	public List<NestingBrick> getAllNestingBrickParts(boolean sorted) {
+		//TODO: handle sorting
+		List<NestingBrick> nestingBrickList = new ArrayList<NestingBrick>();
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/IfLogicEndSimpleBrick.java]: this look a lot like duplicate code

**Comment by bernadettespieler** [comment]
not mergeble

**Pull request closed without merging**